tests/driver_at86rf2xx/recv.c

```diff
@@ -98,7 +98,7 @@ void recv(netdev_t *dev)
     printf("Seq.: %u\n", (unsigned)ieee802154_get_seq(buffer));
     od_hex_dump(buffer + mhr_len, data_len - mhr_len, 0);
     printf("txt: ");
-    for (int i = mhr_len; i < data_len; i++) {
+    for (size_t i = mhr_len; i < data_len; i++) {
```

Comment:
Use rather `unsigned` as a counting variable.

Reply:
The change here is because the comparison shouldn't be made between two variables of different size. `mhr_len` and `data_len` are declared as `size_t`.

Reply:
a) where did you get that from? Comparisons between variables of different size shouldn't be a problem at all
b) I think on all our platforms `sizeof(size_t) == sizeof(unsigned)` is true.

Reply:
> a) where did you get that from? Comparisons between variables of different size shouldn't be a problem at all

I said they _shouldn't_, because it _might_ lead to errors, if a counter variable is smaller than the variable it's compared with of course it cannot reach the required length. It will only work if the count doesn't exceed it's own type length (which yes, it's mostly what happens).

> b) I think on all our platforms sizeof(size_t) == sizeof(unsigned) is true.

I _think_ it's not enough, our code needs to be portable. `sizeof()` returns `size_t`, you can find [here](https://gcc.gnu.org/ml/gcc-help/2006-06/msg00188.html) a small discussion about it.